repos/system_upgrade/el7toel8/actors/checkminimumrequirements/libraries/library.py

```diff
@@ -0,0 +1,89 @@
+import os
+import re
+
+from leapp.models import HardwareInfo
+from leapp.libraries.common.config import architecture
+from leapp.libraries.stdlib import api
+from leapp.reporting import create_report
+from leapp import reporting
+
+min_req_data = {
+    'memory': {
+        architecture.ARCH_X86_64: 1536,  # 1.5G
+        architecture.ARCH_ARM64: 2048,  # 2Gb
+        architecture.ARCH_PPC64LE: 2048,  # 2Gb
+        architecture.ARCH_S390X: 1024  # 1Gb
+    },
+    'kernel_version': '3.10'
+}
+
+
+def _get_kernel_version(hardware_info):
+    hardware_info.kernel_version = re.findall(r'\d+\.\d+\.\d+', os.uname()[2])[0]
+
+
+def _check_kernel_version(minimum_req_error, hardware_info):
+    kernel_ver_info = tuple(map(int, hardware_info.kernel_version.split('.')))
+    if kernel_ver_info < (3, 10, 0):
+        minimum_req_error.append('Minimu kernel version: {}    found: {}'.format(
+            min_req_data['kernel_version'], hardware_info.kernel_version))
+
+
+def _get_memory_info(hardware_info):
+    mem_info = dict((i.split()[0].rstrip(':'), int(i.split()[1])) for i in open('/proc/meminfo').readlines())
+    total_mem_kib = mem_info['MemTotal']
+
+    hardware_info.mem_total = total_mem_kib
+
```

Comment:
obiously these functions cannot be part of this actor. This is `check` actor. That is checking something. It is not scanning the system.  Check actors are supposed to decide / check something based on consumed data.